src/util/p13n.ts

```diff
@@ -0,0 +1,32 @@
+import { isInEditor, isInPreviewAsVisitor } from './editor';
+
+export const getContentVariant = (
+  content: any,
+  templateAnnotations?: Record<string, string>
+) => {
+  if (
+    !content ||
+    !templateAnnotations ||
+    !isInEditor() ||
+    isInPreviewAsVisitor()
+  ) {
+    return content;
+  }
+
+  const variant = templateAnnotations[content['@path']]?.match(
+    /selectedVariant="(.+)"/
+  )?.[1];
+
+  if (!variant || variant === content['@name']) {
+    return content;
+  }
+
+  const contentVariant = content[variant];
+
+  if (!contentVariant) {
+    console.warn(`Missing variant ${variant} for content ${content['@path']}.`);
```

Comment:
friendly warning for if the variant content is absent. this is typically because variants=all is missing as a query parameter when retrieving content from the magnolia rest api.